### ISSUE TITLE
Problem with sending data to LRS, error 500 with "in_array()" (v 1.7.0)

### ISSUE THREAD

**Issue by Lupei** [LearningLocker/learninglocker#701]
Hello,

I try to send data from Articulate Storyline 2 courses to LearningLocker using the endpoint describes on that page : https://www.articulate.com/tincanapi/

When I lauch a courses, I get this error :
{"error":true,"success":false,"message":"in_array() expects parameter 2 to be array, null given","code":500}

I was before on the version 1.4.2 and everthings work right. I tried to jump to the 1.7.0 but then, nothing.

Can you help me ?

**Comment by ryansmith94** [comment]
Hi @Lupei, thanks for reporting the issue. Please could you clear your "app/storage/logs/laravel.log" file, reproduce your issue, then paste the contents of the "app/storage/logs/laravel.log" file here?

**Comment by Lupei** [comment]
Hello @ryansmith94, yes of course. There it is :


[2015-07-29 13:26:49] local.ERROR: exception 'ErrorException' with message 'in_array() expects parameter 2 to be array, null given' in C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentStorer.php:31
Stack trace:
#0 [internal function]: Illuminate\Exception\Handler->handleError(2, 'in_array() expe...', 'C:\\xampp\\htdocs...', 31, Array)
#1 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentStorer.php(31): in_array('all', NULL)
#2 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentRepository.php(31): Locker\Repository\Statement\EloquentStorer->store(Array, Array, Object(Locker\Repository\Statement\StoreOptions))
#3 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementStoreController.php(129): Locker\Repository\Statement\EloquentRepository->store(Array, Array, Array)
#4 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementStoreController.php(86): Controllers\xAPI\StatementStoreController->createStatements(Array, Object(Closure))
#5 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementController.php(63): Controllers\xAPI\StatementStoreController->update(Array)
#6 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\BaseController.php(44): Controllers\xAPI\StatementController->update()
#7 [internal function]: Controllers\xAPI\BaseController->selectMethod()
#8 C:\xampp\htdocs\tincan_lrs\vendor\laravel\framework\src\Illuminate\Routing\Controller.php(231): call_user_func_array(Array, Array)
#9 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5816): Illuminate\Routing\Controller->callAction('selectMethod', Array)
#10 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5804): Illuminate\Routing\ControllerDispatcher->call(Object(Controllers\xAPI\StatementController), Object(Illuminate\Routing\Route), 'selectMethod')
#11 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5003): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'Controllers\\xAP...', 'selectMethod')
#12 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#13 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5362): call_user_func_array(Object(Closure), Array)
#14 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5028): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#15 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5016): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#16 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(717): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#17 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(698): Illuminate\Foundation\Application->dispatch(Object(Illuminate\Http\Request))
#18 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(7738): Illuminate\Foundation\Application->handle(Object(Illuminate\Http\Request), 1, true)
#19 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8341): Illuminate\Session\Middleware->handle(Object(Illuminate\Http\Request), 1, true)
#20 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8288): Illuminate\Cookie\Queue->handle(Object(Illuminate\Http\Request), 1, true)
#21 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(10953): Illuminate\Cookie\Guard->handle(Object(Illuminate\Http\Request), 1, true)
#22 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(659): Stack\StackedHttpKernel->handle(Object(Illuminate\Http\Request))
#23 C:\xampp\htdocs\tincan_lrs\public\index.php(49): Illuminate\Foundation\Application->run()
#24 {main} [] []
[2015-07-29 13:26:55] local.ERROR: exception 'ErrorException' with message 'in_array() expects parameter 2 to be array, null given' in C:\xampp\htdocs\tincan_lrs\app\locker\repository\Document\EloquentDocumentRepository.php:144
Stack trace:
#0 [internal function]: Illuminate\Exception\Handler->handleError(2, 'in_array() expe...', 'C:\\xampp\\htdocs...', 144, Array)
#1 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Document\EloquentDocumentRepository.php(144): in_array('all', NULL)
#2 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Document\EloquentDocumentRepository.php(202): Locker\Repository\Document\EloquentDocumentRepository->filterScopes(Array, 'state')
#3 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Document\EloquentDocumentRepository.php(66): Locker\Repository\Document\EloquentDocumentRepository->findStateDoc(Array, 'resume', 'http://6dKiaVvL...', Array, '760e3480-ba55-4...', true)
#4 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\DocumentController.php(221): Locker\Repository\Document\EloquentDocumentRepository->find(Array, 'state', Array)
#5 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\DocumentController.php(69): Controllers\xAPI\DocumentController->documentResponse(Array)
#6 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\BaseController.php(51): Controllers\xAPI\DocumentController->show()
#7 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\BaseController.php(43): Controllers\xAPI\BaseController->get()
#8 [internal function]: Controllers\xAPI\BaseController->selectMethod()
#9 C:\xampp\htdocs\tincan_lrs\vendor\laravel\framework\src\Illuminate\Routing\Controller.php(231): call_user_func_array(Array, Array)
#10 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5816): Illuminate\Routing\Controller->callAction('selectMethod', Array)
#11 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5804): Illuminate\Routing\ControllerDispatcher->call(Object(Controllers\xAPI\StateController), Object(Illuminate\Routing\Route), 'selectMethod')
#12 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5003): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'Controllers\\xAP...', 'selectMethod')
#13 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#14 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5362): call_user_func_array(Object(Closure), Array)
#15 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5028): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#16 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5016): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#17 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(717): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#18 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(698): Illuminate\Foundation\Application->dispatch(Object(Illuminate\Http\Request))
#19 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(7738): Illuminate\Foundation\Application->handle(Object(Illuminate\Http\Request), 1, true)
#20 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8341): Illuminate\Session\Middleware->handle(Object(Illuminate\Http\Request), 1, true)
#21 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8288): Illuminate\Cookie\Queue->handle(Object(Illuminate\Http\Request), 1, true)
#22 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(10953): Illuminate\Cookie\Guard->handle(Object(Illuminate\Http\Request), 1, true)
#23 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(659): Stack\StackedHttpKernel->handle(Object(Illuminate\Http\Request))
#24 C:\xampp\htdocs\tincan_lrs\public\index.php(49): Illuminate\Foundation\Application->run()
#25 {main} [] []
[2015-07-29 13:26:56] local.ERROR: exception 'ErrorException' with message 'in_array() expects parameter 2 to be array, null given' in C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentStorer.php:31
Stack trace:
#0 [internal function]: Illuminate\Exception\Handler->handleError(2, 'in_array() expe...', 'C:\\xampp\\htdocs...', 31, Array)
#1 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentStorer.php(31): in_array('all', NULL)
#2 C:\xampp\htdocs\tincan_lrs\app\locker\repository\Statement\EloquentRepository.php(31): Locker\Repository\Statement\EloquentStorer->store(Array, Array, Object(Locker\Repository\Statement\StoreOptions))
#3 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementStoreController.php(129): Locker\Repository\Statement\EloquentRepository->store(Array, Array, Array)
#4 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementStoreController.php(86): Controllers\xAPI\StatementStoreController->createStatements(Array, Object(Closure))
#5 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\StatementController.php(63): Controllers\xAPI\StatementStoreController->update(Array)
#6 C:\xampp\htdocs\tincan_lrs\app\controllers\xapi\BaseController.php(44): Controllers\xAPI\StatementController->update()
#7 [internal function]: Controllers\xAPI\BaseController->selectMethod()
#8 C:\xampp\htdocs\tincan_lrs\vendor\laravel\framework\src\Illuminate\Routing\Controller.php(231): call_user_func_array(Array, Array)
#9 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5816): Illuminate\Routing\Controller->callAction('selectMethod', Array)
#10 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5804): Illuminate\Routing\ControllerDispatcher->call(Object(Controllers\xAPI\StatementController), Object(Illuminate\Routing\Route), 'selectMethod')
#11 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5003): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'Controllers\\xAP...', 'selectMethod')
#12 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#13 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5362): call_user_func_array(Object(Closure), Array)
#14 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5028): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#15 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(5016): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#16 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(717): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#17 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(698): Illuminate\Foundation\Application->dispatch(Object(Illuminate\Http\Request))
#18 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(7738): Illuminate\Foundation\Application->handle(Object(Illuminate\Http\Request), 1, true)
#19 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8341): Illuminate\Session\Middleware->handle(Object(Illuminate\Http\Request), 1, true)
#20 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(8288): Illuminate\Cookie\Queue->handle(Object(Illuminate\Http\Request), 1, true)
#21 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(10953): Illuminate\Cookie\Guard->handle(Object(Illuminate\Http\Request), 1, true)
#22 C:\xampp\htdocs\tincan_lrs\bootstrap\compiled.php(659): Stack\StackedHttpKernel->handle(Object(Illuminate\Http\Request))
#23 C:\xampp\htdocs\tincan_lrs\public\index.php(49): Illuminate\Foundation\Application->run()
#24 {main} [] []

**Comment by daniel-abbey** [comment]
Hi @Lupei, please can you paste the PUT request Storyline is sending to the statements API so I can try to recreate the error. Thanks

**Comment by Lupei** [comment]
Hello @daniel-abbey, here's the header : 

Host: srv.advensys.fr
User-Agent: Mozilla/5.0 (Windows NT 6.3; WOW64; rv:39.0) Gecko/20100101 Firefox/39.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: fr,fr-FR;q=0.8,en-US;q=0.5,en;q=0.3
Accept-Encoding: gzip, deflate
Referer: http://srv.advensys.fr/tincan_lms/webroot/cours/M01L07%20-%20Demander%20des%20informations%20-%205%20-%20Storyline%20output/story.html?endpoint%3Dhttp%3A%2F%2Fsrv.advensys.fr%2Ftincan_lrs%2Fpublic%2Fdata%2FxAPI%2F&auth%3DBasic%20M2MzNTc0NGQ1MzQ2MWY0MjNmN2U0NjI0NGEwZGY1NWVmYjBmMDExODo3ODYwNmRjZjc4NDc0YzkxNWJhMWVjOGNkYzI0ZTNmMzFkZDFkMjY5&actor%3D%7B%22name%22%3A%20%5B%22admin%20admin%22%5D%2C%20%22mbox%22%3A%20%5B%22mailto%3Aadmin@advensys.fr%22%5D%7D&registration%3D760e3480-ba55-4991-94b0-000000000020
Content-Length: 1203
Content-Type: text/plain; charset=UTF-8
Cookie: laravel_session=eyJpdiI6Ikh3REk3dUFYSE9PTkRSbVc2Q3pqTngxNGZSbUh3V3lwSDluYnllTWpIUkE9IiwidmFsdWUiOiJoZ3Z5OU9lXC8wQU16UEJMdUFDaTZiWGNJR0o3YnRCU2pCXC9CbmR0MDFkeDFHWVBCSldVSHBEWVFWejUyNnZ1NVl1cDNRc2dUaU9vN3BQdklma25OZlB3PT0iLCJtYWMiOiJmYzVhNzNjZmU0NTMyNTNlZWQ0N2Q4NGQ5YjdhNTI4NDllYWU1MWI5ZmE1YmU2N2E1YWM4YmJiNmQzYjE5MGVjIn0%3D
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache

And the body : 

Authorization=Basic%20M2MzNTc0NGQ1MzQ2MWY0MjNmN2U0NjI0NGEwZGY1NWVmYjBmMDExODo3ODYwNmRjZjc4NDc0YzkxNWJhMWVjOGNkYzI0ZTNmMzFkZDFkMjY5&X%2DExperience%2DAPI%2DVersion=1%2E0%2E1&statementId=c8bfe27c%2D538f%2D47b1%2D97d0%2Da9c124adc7d8&Content%2DType=application%2Fjson&content=%7B%22actor%22%3A%7B%22mbox%22%3A%22mailto%3Aadmin%40advensys%2Efr%22%2C%22name%22%3A%22admin%20admin%22%7D%2C%22verb%22%3A%7B%22id%22%3A%22http%3A%2F%2Fadlnet%2Egov%2Fexpapi%2Fverbs%2Fattempted%22%7D%2C%22context%22%3A%7B%22contextActivities%22%3A%7B%22grouping%22%3A%7B%22id%22%3A%22http%3A%2F%2F6UKMIZD8iV2%5Fcourse%5Fid%22%7D%2C%22parent%22%3A%7B%22id%22%3A%22http%3A%2F%2F6UKMIZD8iV2%5Fcourse%5Fid%22%7D%7D%2C%22registration%22%3A%22760e3480%2Dba55%2D4991%2D94b0%2D000000000020%22%7D%2C%22result%22%3A%7B%22completion%22%3Afalse%7D%2C%22version%22%3A%221%2E0%2E0%22%2C%22object%22%3A%7B%22definition%22%3A%7B%22name%22%3A%7B%22und%22%3A%22M01L07%20%2D%20Demander%20des%20informations%20%2D%205%22%7D%2C%22type%22%3A%22http%3A%2F%2Fadlnet%2Egov%2Fexpapi%2Factivities%2Fcourse%22%2C%22description%22%3A%7B%22und%22%3A%22%22%7D%7D%2C%22id%22%3A%22http%3A%2F%2F6UKMIZD8iV2%5Fcourse%5Fid%22%2C%22objectType%22%3A%22Activity%22%7D%7D

**Comment by daniel-abbey** [comment]
Hi @Lupei, thanks for that. Just to make sure, could you let me know the URL too?

**Comment by Lupei** [comment]
Hello @daniel-abbey, i'm not sure it was this couses because I have many, but here's a link :

http://srv.advensys.fr/tincan_lms/webroot/cours/M01E2%20-%20Prendre%20une%20reservation%20-%20Storyline%20output/story.html?endpoint%3Dhttp%3A%2F%2Fsrv.advensys.fr%2Ftincan_lrs%2Fpublic%2Fdata%2FxAPI%2F&auth%3DBasic%20M2MzNTc0NGQ1MzQ2MWY0MjNmN2U0NjI0NGEwZGY1NWVmYjBmMDExODo3ODYwNmRjZjc4NDc0YzkxNWJhMWVjOGNkYzI0ZTNmMzFkZDFkMjY5&actor%3D%7B"name"%3A%20%5B"admin%20admin"%5D%2C%20"mbox"%3A%20%5B"mailto%3Aadmin@advensys.fr"%5D%7D&registration%3D760e3480-ba55-4991-94b0-000000000021

**Comment by ryansmith94** [comment]
Hi @Lupei, @daniel-abbey meant the URL of the [request that you pasted earlier](https://github.com/LearningLocker/learninglocker/issues/701#issuecomment-125980507).

**Comment by Lupei** [comment]
Hello @ryansmith94, oh I see. My bad

There it is : 

http://srv.advensys.fr/tincan_lrs/public/data/xAPI/statements?method=PUT

**Comment by ryansmith94** [comment]
Awesome cheers @Lupei :+1:

**Comment by daniel-abbey** [comment]
@Lupei please can you confirm you followed the [upgrade instructions on the LL site](http://docs.learninglocker.net/upgrading/)?

**Comment by Lupei** [comment]
Hello @daniel-abbey. I didn't upgrade, I made a fully new install

**Comment by daniel-abbey** [comment]
@Lupei did you have existing content and users or did you create these after the new install?

**Comment by Lupei** [comment]
@daniel-abbey I had, I keep the database I created with the previous version

**Comment by daniel-abbey** [comment]
@Lupei you may need to migrate the database using the command `php artisan migrate --env=YOUR_ENV` where `YOUR_ENV` should be replaced with your environment (likely to be local or production).

**Comment by Lupei** [comment]
@daniel-abbey I did it, but it broke the LRS. Here's the message :

Status: 404 Error: Error in exception handler: Trying to get property of non-object (View: C:\xampp\htdocs\tincan_lrs\app\views\system\navbar.blade.php) (View: C:\xampp\htdocs\tincan_lrs\app\views\system\navbar.blade.php) (View: C:\xampp\htdocs\tincan_lrs\app\views\system\navbar.blade.php) in C:\xampp\htdocs\tincan_lrs\app\locker\helpers\Access.php:13

**Comment by daniel-abbey** [comment]
@Lupei, when did you encounter this error? It is unrelated to the database migration process.

**Comment by Lupei** [comment]
@daniel-abbey however I had this error juste after doing the migration, before that I could access to the LRS website without problem

**Comment by daniel-abbey** [comment]
@Lupei do you have the previous LL 1.4.2 available? I would recommend upgrading that version rather than a fresh install.

**Comment by Lupei** [comment]
@daniel-abbey No I don't have the previous version anymore unfortunatly. Thanks for the help, I will use a whole new install

**Comment by daniel-abbey** [comment]
@Lupei, I have completed the following in an attempt to reproduce your issue:

1. install v1.4.2
2. generate statements with Golf Example from tincanapi.com/prototypes
3. access the statements from current v1.7.0 with config set to use database created by v1.4.2

Unfortunately I did not encounter the issue you did and therefore was unable to reproduce the error. Bear in mind I was unable to emulate a Storyline 2 environment.

Are you happy for this issue to be closed?
